src/shared/modules/features/versionedFeatures.ts

```diff
@@ -31,70 +32,71 @@ export const FIRST_MULTI_DB_SUPPORT = NEO4J_4_0
 // compatible bolt server.
 export const FIRST_NO_MULTI_DB_SUPPORT = '3.4.0'
 
-export const canSendTxMetadata = (state: any) => {
+export const canSendTxMetadata = (state: GlobalState) => {
   const serverVersion = guessSemverVersion(getVersion(state))
 
-  if (!semver.valid(serverVersion)) {
+  if (!serverVersion || !semver.valid(serverVersion)) {
     return false
+  } else {
+    return semver.gt(serverVersion, NEO4J_TX_METADATA_VERSION)
   }
-  return semver.gt(serverVersion, NEO4J_TX_METADATA_VERSION)
 }
 
-export const getShowCurrentUserProcedure = (serverVersion: any) => {
-  serverVersion = guessSemverVersion(serverVersion)
+export const getShowCurrentUserProcedure = (serverVersion: string) => {
+  const serverVersionGuessed = guessSemverVersion(serverVersion)
 
   const pre4 = 'CALL dbms.security.showCurrentUser()'
-  if (!semver.valid(serverVersion)) {
+  if (!semver.valid(serverVersionGuessed)) {
     return pre4
   }
-  if (semver.gte(serverVersion, NEO4J_4_0)) {
+  if (serverVersionGuessed && semver.gte(serverVersionGuessed, NEO4J_4_0)) {
     return 'CALL dbms.showCurrentUser()'
   }
   return pre4
 }
 
-export const getDbClusterRole = (state: any) => {
+export const getDbClusterRole = (state: GlobalState) => {
   const pre4 = 'CALL dbms.cluster.role() YIELD role'
   const serverVersion = guessSemverVersion(getVersion(state))
   if (!semver.valid(serverVersion)) {
     return pre4
   }
-  if (semver.gte(serverVersion, NEO4J_4_0)) {
+  if (serverVersion && semver.gte(serverVersion, NEO4J_4_0)) {
     const db = getUseDb(state)
     return `CALL dbms.cluster.role("${db}") YIELD role`
   }
   return pre4
 }
 
-export const hasMultiDbSupport = (state: any) => {
+export const hasMultiDbSupport = (state: GlobalState) => {
   const serverVersion = guessSemverVersion(getVersion(state))
   if (!semver.valid(serverVersion)) {
     return false
   }
-  if (semver.gte(serverVersion, NEO4J_4_0)) {
+  if (serverVersion && semver.gte(serverVersion, NEO4J_4_0)) {
     return true
   }
   return false
 }
 
-export const getUsedDbName = (state: any) => {
+export const getUsedDbName = (state: GlobalState) => {
   const serverVersion = guessSemverVersion(getVersion(state))
   if (!semver.valid(serverVersion)) {
     return undefined
   }
-  if (semver.gte(serverVersion, NEO4J_4_0)) {
+  if (serverVersion && semver.gte(serverVersion, NEO4J_4_0)) {
     return getUseDb(state)
   }
   return getActiveDbName(state)
 }
 
-export const getDefaultBoltScheme = (serverVersion: any) => {
-  serverVersion = guessSemverVersion(serverVersion)
+export const getDefaultBoltScheme = (serverVersion: string | null) => {
+  const serverVersionGuessed = guessSemverVersion(serverVersion)
   const pre4 = 'bolt://'
```

Comment:
It's not related to the PR. But can you try to rename it to a more descriptive one?

Reply:
Which one do you mean, the pre4 variable?

Reply:
@eijawerner - yes!

Reply:
As it's unrelated to the main PR, it seems to me this shouldn't hold the PR from being merged? Seeing how we want to get a release candidate out soon. Also do you have any suggestion for a better name?

Reply:
@OskarDamkjaer - agree. I'd say we can rename it as `prefix` from this context.

Reply:
Is it ok if I leave it as it is for this PR? I think it is ok even if it is not perfect (and the pattern is used in several places in the file), and also as you said it is not really part of this PR.